icu4c/source/i18n/unicode/timezone.h

```diff
@@ -444,6 +444,31 @@ class U_I18N_API TimeZone : public UObject {
     static UnicodeString& U_EXPORT2 getCanonicalID(const UnicodeString& id,
         UnicodeString& canonicalID, UBool& isSystemID, UErrorCode& status);
 
+
+
+    /**
+     * Returns the preferred time zone ID in the IANA time zone database for the given time zone ID.
+     * There are two types of preferred IDs. The first type is the one defined in zone.tab file,
+     * such as "America/Los_Angeles". The second types is the one defined for zones not associated
+     * with a specific region, but not defined with "Link" syntax such as "Etc/GMT+10".
+     *
+     * <p>Note: For most of system time zone IDs, this method returns an ID same as getCanonicalID().
+     * getCanonicalID() is based on canonical time zone IDs defined in Unicode CLDR.
+     * These canonical time zone IDs in CLDR were based on very old version of the time zone database.
+     * In the IANA time zone database, some IDs were updated since then. This API returns a newer
+     * time zone ID. For example, CLDR defines "Asia/Calcutta" as the canonical time zone ID. This
+     * method returns "Asia/Kolkata" instead.
+     *
+     * @param id        The input time zone ID.
+     * @param ianaID    Receives the preferred time zone ID in the IANA time zone database.
+     * @param status    Receives the status.  When the given time zone ID is not a known system time zone
+     *                  ID, U_ILLEGAL_ARGUMENT_ERROR is set.
+     * @return  A reference to the result.
```

Comment:
Should we also document that the return value could be bogus for error and "" if the input is "Etc/Unkown" ? or should both case return bogus string ?

Reply:
I will add a note in the API reference, explaining about "Etc/Unknown".

Reply:
Since this has a UErrorCode parameter, not sure we want to make setting return to bogus part of the API contract.

Reply:
done